### PR TITLE
Add Hack for Generating CRDs from external sources

### PR DESCRIPTION
**Pull request by t-eckert**
For the Gateway API implementation to work, the target Kubernetes cluster must have the relevant CRDs installed. These CRDs are managed by the SIG Network group. We'll want to generate copies of these CRDs and store them in the `/crds` directory in our Helm chart so they can be installed for our users automatically.

This PR adds a `kustomization.yaml` file for managing external dependencies and a `generate-external-crds` command which will load in the dependencies and nicely format them in our `/crds` directory.

This will make it easy for us to keep the CRDs up to date and allows for extensibility if we ever need to add another set of external CRDs to the project in the future.

Changes proposed in this PR:
- Add `generate-external-crds`

How I've tested this PR:
- I ran the hack and it generated the correct CRDs which allowed them to be installed automatically when I installed Consul onto Kubernetes.

How I expect reviewers to test this PR:
- You may also run the hack. 

Completes https://hashicorp.atlassian.net/browse/NET-3663 

Checklist:
~- [ ] Tests added~
~- [ ] CHANGELOG entry added~
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: hack/generate-external-crds/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v0.6.2
```
Comment (t-eckert) [inline hack/generate-external-crds/kustomization.yaml:5]: In the future, we can just change the values in this list and re-run the script to update CRDs.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,6 +24,7 @@
     1. [Writing Acceptance tests](#writing-acceptance-tests)
 1. [Using the Acceptance Test Framework to Debug](#using-acceptance-test-framework-to-debug)
 1. [Helm Reference Docs](#helm-reference-docs)
+1. [Managing External CRD Dependencies](#managing-external-crd-dependencies)
```
Comment (thisisnotashwin) [inline CONTRIBUTING.md:27]: quick drive by question 🚗: Currently, these CRDs will always get installed with a helm install. We should add an `if` condition around them as we do with the other CRDs to ensure they only get installed on a cluster if the gateway is enabled. Was this change intentional?

Comment (thisisnotashwin) [reply CONTRIBUTING.md:27]: I guess it is ok for this PR since we arent merging it into `main` but it is definitely something we should consider before that happens. This looks really clever otherwise. Nicely done.

File: charts/consul/crds/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (lkysow) [inline charts/consul/crds/kustomization.yaml:1]: This shouldn't be in here I don't think

File: CONTRIBUTING.md (hunk 2 above)
Comment (lkysow) [reply CONTRIBUTING.md:27]: This is using the Helm CRDs directory, not our normal CRDs method. That directory doesn't support templating.

File: charts/consul/crds/kustomization.yaml (hunk 3 above)
Comment (lkysow) [reply charts/consul/crds/kustomization.yaml:1]: Oh I see this is being helm ignored. Can you add a comment at the top that this is helm ignored and only used for the make <whatever it is> command?

File: Makefile [hunk 4]
```diff
@@ -123,6 +127,7 @@ lint: cni-plugin-lint ## Run linter in the control-plane, cli, and acceptance di
 ctrl-manifests: get-controller-gen ## Generate CRD manifests.
 	cd control-plane; $(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
 	make copy-crds-to-chart
+	make generate-example-crds
```
Comment (lkysow) [inline Makefile:130]:
```suggestion
	make generate-external-crds
```

**Review by lkysow — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (thisisnotashwin) [reply CONTRIBUTING.md:27]: Aaha! so this needs an explicit install! thanks for the clarification!

Comment (t-eckert) [reply CONTRIBUTING.md:27]: Good call out though! It's a pain and I tried to get around it, but there really wasn't a way to do it :shrug:

File: Makefile (hunk 4 above)
Comment (t-eckert) [reply Makefile]: 🤦🏻

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1216,6 +1217,26 @@ So that the documentation can look like:
 - `ports` ((#v-ingressgateways-defaults-service-ports)) (`array<map>: [{port: 8080, port: 8443}]`) - Port docs
 ```
 
+## Managing External CRD Dependencies
+
+Some of the features of Consul on Kubernetes make use of custom resource definitions (CRDs) that we don't directly
```
Comment (nathancoleman) [inline CONTRIBUTING.md:1222]:
```suggestion
Some of the features of Consul on Kubernetes make use of Custom Resource Definitions (CRDs) that we don't directly
```

**Review by nathancoleman — APPROVED** [review]

**Review by andrewstucki — APPROVED** [review]
